compiler/rustc_mir/src/util/pretty.rs

```diff
@@ -408,6 +409,33 @@ impl ExtraComments<'tcx> {
     }
 }
 
+fn use_verbose(ty: &&TyS<'tcx>) -> bool {
+    match ty.kind() {
+        ty::Int(_) | ty::Uint(_) | ty::Bool | ty::Char | ty::Float(_) => false,
+        // Unit type
+        ty::Tuple(g_args) if g_args.is_empty() => false,
+        ty::Tuple(g_args) => {
+            // could have used `try_fold` here but it seems a bit silly that
+            // the accumulator is useless
```

Comment:
I believe you could write the entire thing as a single `any` call on `g_args.iter()`.